File: Client/Android/solutions/rtc-api-example/rtc-api-example/src/main/java/com/vertc/api/example/utils/IMEUtils.java
package com.vertc.api.example.utils;

import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import androidx.annotation.NonNull;

public class IMEUtils {
    public static void closeIME(@NonNull View v) {
        InputMethodManager imm = (InputMethodManager) v.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
    }
}

File: Client/Android/component/solution-base/src/main/java/com/vertcdemo/core/utils/IMEUtils.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
package com.vertcdemo.core.utils

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

object IMEUtils {
    @JvmStatic
    fun closeIME(v: View) {
        val mgr = v.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        mgr.hideSoftInputFromWindow(v.windowToken, 0)
        v.clearFocus()
    }

    @JvmStatic
    fun openIME(v: EditText) {
        v.requestFocus()
        if (v.hasFocus()) {
            v.post {
                val mgr =
                    v.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                mgr.showSoftInput(v, InputMethodManager.SHOW_FORCED)
            }
        }
    }
}
